File: hard/security/Security.java
import java.util.Scanner;

public class Security {
    public static void main(final String[] args) {
        var floor = new Scanner(System.in).nextLine();
        var isSafe = floor.matches(".*[$T].*G.*[$T].*");

        System.out.println(isSafe ? "quiet" : "ALARM");
    }
}

File: hard/security/Security.c
#include <stdio.h>
#define MAX 200

int main() {
  char floor[MAX];
  char *message;
  int alarm = 0;
  fgets(floor, MAX, stdin);
  
  for (char *c = floor; *c && !message; ++c) 
    if (*c == 'G' && alarm) message = "quiet";
    else if ((*c == '$' || *c == 'T') && alarm++) message = "ALARM"; 
  	   
  puts(message);
  return 0;
}
